app/models/state_change.rb

```diff
@@ -11,47 +13,55 @@ class StateChange < ApplicationRecord
   include Uuid::Uuidable
   include Asset::Ownership::ChangesOwner
 
-  belongs_to :user
-  validates :user, presence: true
+  attr_accessor :customer_accepts_responsibility
 
+  belongs_to :user, optional: false
   # This is the target asset for which to update the state
-  belongs_to :target, class_name: 'Labware'
-  validates :target, presence: true
-
-  # If the state change is a known failure state then a reason must be included
-  validates :reason, presence: true, if: :targetted_for_failure?
-
-  def targetted_for_failure?
-    %w[failed cancelled].include?(target_state)
-  end
-  private :targetted_for_failure?
-
-  set_target_for_owner(:target)
+  belongs_to :target, class_name: 'Labware', optional: false
 
   # Some targets can have "contents" updated (notably plates).  The meaning of this is is dealt with by the
   # target being updated.
   serialize :contents
 
-  attr_accessor :customer_accepts_responsibility
-
+  # If the state change is a known failure state then a reason must be included
+  validates :reason, presence: true, if: :targetted_for_failure?
   # These track the state of the target.  The target_state is what we want it to end up in and the previous_state
   # records the state that it was in before the update.  The previous_state is not assigned by the creator but
   # by the action of making the transition.
   validates :target_state, presence: true
-  validates_unassigned(:previous_state)
+  validates_unassigned :previous_state
+  # If we don't have a state changer configured, we can't change the state
+  validates :state_changer, presence: { message: 'target does not have a configured state changer' }
 
-  # Before creating an instance we record the current state of the target.
   before_create :record_current_state_of_target
+  after_create :update_state_of_target
+
+  set_target_for_owner(:target)
+
+  private
+
+  def targetted_for_failure?
+    %w[failed cancelled].include?(target_state)
+  end
+
+  # Before creating an instance we record the current state of the target.
   def record_current_state_of_target
     self.previous_state = target.state
   end
-  private :record_current_state_of_target
+
+  def state_changer
+    target.state_changer
+  end
 
   # After creation update the state of the target asset, leaving it to do the right thing.
   # After state change, update the owner
-  after_create :update_state_of_target
   def update_state_of_target
```

Comment:
Main change. Use our state changers. Allows us to remove transition_to from a whole host of places, and centralise state transitions.